### PR TITLE
Add Pagination to Evolving Conditions Table (#225)

### LINKED ISSUE #225: Page Evolving Conditions Table
**Is your feature request related to a problem? Please describe.**
The evolving conditions table can grow quite long. For example, the Chapman mechanism example is so long that the page lags.

To see where this should happen

1. Go to [musicbox.acom.ucar.edu](https://musicbox.acom.ucar.edu/)
2. Click on Get Started
3. Click on Select Example
4. Click on Chapman Cycle
5. Click on Conditions in the left side bar
6. Click on the Evolving tab

**Describe the solution you'd like**
The evolving conditions table should be paged.

Screenshot 2024-01-04 at 2 32 54 PM

# Acceptance Criteria

- The evolving conditions table is paged. 
- The number of items shown per page can be configured

**Files involved**
- `src/components/Conditions/EvolvingConditionsTab.jsx`
- `src/components/Conditions/EvolvingConditionsDetail.jsx`

To do this issue, you'll need to connect your local version to our production server or setup a dev api server for yourself. See our readme for details on both

### PR DESCRIPTION
**Pull request by ketansahugit**
## Pull Request Summary

This pull request addresses issue #225 

### Changes Made

- Added pagination to the evolving conditions table on the 'Evolving' tab.
- Implemented the ability to configure the number of items shown per page.

Closes #225

### CODE REVIEW

**Review by K20shores — CHANGES_REQUESTED** [review]
Well done! I tested it out and it seems to work flawlessly; I left two comments. One related to styling, and another to use the react bootstrap components for pagination.

As it stands, this PR is great! If you don't have time to address the comments, I completely understand, but I would like these changes if possible.

Either way, thank you so much for this contribution!

File: src/components/Conditions/EvolvingConditionsDetail.jsx [hunk 1]
```diff
@@ -45,6 +67,22 @@ const EvolvingConditionsDetail = (props) => {
...
+            <select
+              className="mx-2"
+              value={itemsPerPage}
+              onChange={(e) => handleItemsPerPageChange(Number(e.target.value))}
+            >
+              <option value={50}>50</option>
+              <option value={100}>100</option>
+              <option value={200}>200</option>
+            </select>
+           </label>
```
Comment (K20shores) [inline src/components/Conditions/EvolvingConditionsDetail.jsx:83]: Would it be possible to add some styling? Either some slight internal padding and rounded edges, or maybe using the [react bootstrap select component](https://react-bootstrap.netlify.app/docs/forms/select/).

**Comment by ketansahugit** [comment]
@K20shores Hey, I was about to work on the changes you had suggested, why did you merge the PR?

**Comment by K20shores** [comment]
@ketansahugit because I liked your work! 

I would like the changes I asked for, but I also felt bad asking for them since you're doing this work of your own volition. Feel free to make those changes if you feel inclined and submit another PR though!

**Pull request merged**